wasmtime/_memory.py

```diff
@@ -62,6 +64,57 @@ def data_ptr(self, store: Storelike) -> "ctypes._Pointer[c_ubyte]":
         """
         return ffi.wasmtime_memory_data(store._context, byref(self._memory))
 
+    def read(
+            self,
+            store: Storelike,
+            start: typing.Optional[int] = 0,
+            stop: typing.Optional[int] = None) -> typing.Any:
+        """
+        provide fast way to read large memory slice similar to list[start:stop]
+        negative start, stop is allowed in a way similat to list slice mylist[-10:]
+        """
+        data_ptr = self.data_ptr(store)
+        size = self.data_len(store)
+        key = slice(start, stop, None)
+        start, stop, _ = key.indices(size)
+        val_size = stop - start
+        if val_size <= 0:
+            # return bytearray of size zero
+            return bytearray(0)
+        ptr_type = ctypes.c_ubyte * val_size
+        src_ptr = (ptr_type).from_address(ctypes.addressof(data_ptr.contents) + start)
+        return bytearray(src_ptr)
+
+    def write(
+            self,
+            store: Storelike,
+            value: typing.Any,
+            start: typing.Optional[int] = 0,
+            stop: typing.Optional[int] = None) -> typing.Any:
```

Comment:
Like above, could `value` and the return type here be typed as `bytearray`?

Reply:
Also, could `stop` be omitted here since it's implicitly the length of the `value` array?

Reply:
> Also, could stop be omitted here since it's implicitly the length of the value array?

stop can be used to slice into value without specifying the start point
for example write last 10 bytes of value

```python
memory.write(store, value, None, -10)
memory.write(store, value, stop=-10)
```